nox/manifest.py

```diff
@@ -172,9 +172,23 @@ def filter_by_keywords(self, keywords: str) -> None:
                 session names are checked against.
         """
         self._queue = [
-            x for x in self._queue if keyword_match(keywords, x.signatures + [x.name])
+            x
+            for x in self._queue
+            if keyword_match(keywords, x.signatures + x.tags + [x.name])
         ]
 
+    def filter_by_tags(self, tags: str) -> None:
+        """Filter sessions using pytest-like tag expressions.
+
+        Args:
+            tags (str): A Python expression of tags which session names
+                are checked against.
+
+        Raises:
+            SyntaxError: If the tag expression is invalid.
+        """
+        self._queue = [x for x in self._queue if set(x.tags).intersection(tags)]
+
```

Comment:
Is tags really just a string here? Everywhere else it seems to be `list[str]`? If it is a string, the expression `set(x.tags).intersection(<a_string>)` will always evaluate to an empty set (falsy):

<img width="657" alt="image" src="https://user-images.githubusercontent.com/62767721/174035237-a327cf9b-3371-4ae2-9a1c-856af79d0615.png">


Also AFAICT this would not raise a SyntaxError?

Reply:
Ah, good catch! Yeah, this type annotation should be `list[str]`. I guess `mypy` didn't catch it because `Namespace.tags` doesn't have a static type.